fhir-search/src/main/java/com/ibm/fhir/search/parameters/InclusionParameter.java

```diff
@@ -78,13 +100,23 @@ public boolean equals(Object obj) {
         } else if (!searchParameterTargetType.equals(other.searchParameterTargetType)) {
             return false;
         }
+        if (modifier == null) {
+            if (other.modifier != null) {
+                return false;
+            }
+        } else if (!modifier.equals(other.modifier)) {
+            return false;
+        }
+        if (userSpecifiedTargetType != other.userSpecifiedTargetType) {
+            return false;
+        }
         return true;
     }
 
     @Override
     public String toString() {
         return "InclusionParameter [joinResourceType=" + joinResourceType + ", searchParameter=" + searchParameter + ", searchParameterTargetType="
-                + searchParameterTargetType + "]";
+                + searchParameterTargetType + ", modifier=" + modifier.toString() + ", userSpecifiedTargetType=" + userSpecifiedTargetType + "]";
```

Comment:
Change modifier.toString() to modifier since it can be null.

Reply:
good catch - done.